cwd/editor.js

```diff
@@ -1,3 +1,60 @@
+const update = (btn) => {
+  const viewNum = btn.id.substring(8, 9);
+  const gaugeNum = btn.id.substring(10, 11);
+  const msgNum = btn.id.substring(12, );
+
+  const prob = [];
+  for (const x of Array(5).keys()) {
+    prob.push($('#prob-' + (btn.id.substring(8, )) + "-" + (x + 1)).val());
+  }
+
+  const settings = {
+    "url": views[viewNum - 1] + "gauges/" + gaugeNum + "/messages/" + msgNum,
+    "type": "POST",
+    "timeout": 0,
+    "headers": {
+      "Content-Type": "application/x-www-form-urlencoded"
+    },
+    "data": {
+      "message": "\"" + $('#input-' + (btn.id.substring(8, ))).val() + "\"",
+      "probability": `\[${prob}\]`,
+      "pass": "\"" + $('#passIn').val() + "\""
+    }
+  };
+
+  $.ajax(settings).done(function(response) {
+    console.log(response);
+  });
+}
+
+const updateView = (btn) => {
+  const viewNum = btn.id.substring(8, 9);
+  const msgNum = btn.id.substring(10, 11);
+
+  const prob = [];
+  for (const x of Array(1).keys()) {
+    prob.push($('#prob-' + (btn.id.substring(8, )) + "-" + (x + 1)).val());
+  }
+
+  const settings = {
+    "url": views[viewNum - 1] + "gauges/0/messages/" + msgNum,
```

Comment:
If this is updating a view, why does it update a gauge's messages?

Reply:
It is not updating a gauge's messages. `gauges/0/` is not a real gauge (since we're 1 indexing, this equates to gauge -1 which obviously doesn't exist). This is just included as this is how I'm handling making the view and gauge messages handled by only a single route in the backend (as you requested). If the  backend sees that the gauge is 0 (or really, -1 after adjusting the index) it knows it's editing a view message instead of a gauge message.

Reply:
This was a big oops that we have figured out, so resolving.